src/anndata/_core/extensions.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import inspect
+
+# from collections.abc import Callable
+from pathlib import Path
+from typing import TYPE_CHECKING, Generic, TypeVar
+from warnings import warn
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+from anndata import AnnData
+
+# Based off of the extension framework in Polars
+# https://github.com/pola-rs/polars/blob/main/py-polars/polars/api.py
+
+__all__ = ["register_anndata_namespace"]
+
+
+def find_stacklevel() -> int:
+    """
+    Find the first place in the stack that is not inside AnnData.
+
+    Taken from:
+    https://github.com/pandas-dev/pandas/blob/ab89c53f48df67709a533b6a95ce3d911871a0a8/pandas/util/_exceptions.py#L30-L51
+    """
+    import anndata as ad
```

Comment:
Why an internal-to-the-function import?

Reply:
No reason in particular, it was just what Polars used. As long as the function gets the location of the `anndata/__init__.py` file it'll work. I'll move it outside.

Reply:
It is changed, I moved it outside the function in one of the later pushes.